Server/mods/deathmatch/logic/Utils.cpp

```diff
@@ -526,18 +526,97 @@ bool XMLColorToInt(const char* szColor, unsigned long& ulColor)
 
 bool XMLColorToInt(const char* szColor, unsigned char& ucRed, unsigned char& ucGreen, unsigned char& ucBlue, unsigned char& ucAlpha)
 {
-    // Convert it to an integer first
-    unsigned long ulColor;
-    if (!XMLColorToInt(szColor, ulColor))
-    {
+    // If we're empty, let's just stop right away
+    if (!szColor || strlen(szColor) == 0)
         return false;
+
+    std::vector<SColorRGBA> vecColors;
+    unsigned char           ucCount;
+    if (ColorStringToRGBA(szColor, SColorRGBA(ucRed, ucGreen, ucBlue, ucAlpha), vecColors, ucCount))
+    {
+        ucRed = vecColors[0].R;
+        ucGreen = vecColors[0].G;
+        ucBlue = vecColors[0].B;
+        ucAlpha = vecColors[0].A;
+        return true;
+    }
+    return false;
+}
+
+bool ColorStringToRGBA(const char* szColor, SColorRGBA defaultColor, std::vector<SColorRGBA>& vecColors, unsigned char& ucCount, bool bIgnoreAlpha)

```

Comment:
Please can you explain what "mixed colours" are? This should land in a separate pull request as it's not directly related to supporting more map attributes.